### PR TITLE
Completely avoid instrumenting the PSR NullLogger

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

This is implemented by adding the ability to remove a hook from selected classes directly.

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: zend_abstract_interface/hook/hook.c [hunk 1]
```diff
@@ -226,6 +231,16 @@ static void zai_hook_hash_destroy(zval *zv) {
     efree(hooks);
 }
 
+static inline bool zai_hook_is_excluded(zai_hook_t *hook, zend_class_entry *ce) {
+    if (hook->exclusions.arData && ce) {
+        zend_string *lower = zend_string_tolower(ce->name);
+        bool excluded = zend_hash_exists(&hook->exclusions, lower);
```
Comment (PROFeNoM) [inline zend_abstract_interface/hook/hook.c:237]: That means we only strictly check for the equality of names, which means a class extending `NullLogger` would also get hooked.

I think we do have to iterate through the interfaces `ce->interfaces` + the parents as well

Comment (PROFeNoM) [reply zend_abstract_interface/hook/hook.c:237]: In the .phpt, I think that if we add a class `GrandChild` inherits `Child`, it would get hooked

Comment (bwoebi) [reply zend_abstract_interface/hook/hook.c:237]: Okay, it's a bit more complicated with interfaces: Every class implements all the interfaces any of its parents implement.

The underlying problem is, given a hook on X::foo():
```
interface X { function foo(); }
class A implements X {  function foo() { ... } }
class B extends A { function foo() { ... } }
class C extends B implements X {}
```

I think we agree that if a class (here: B) is directly implementing A, it probably shouldn't be hooked if A::foo() is dropped.
However, now C implements X directly. Should or should it not inherit? The catch though here is that PHP does not expose whether the interface is inherited from the parent only or directly inherited by the child.

I think it's fine to ignore "implements directly".

**Review by PROFeNoM — APPROVED** [review]
Nice 😃

**Pull request merged**